### PR TITLE
Move to Java 17 and higher

### PR DESCRIPTION
**Pull request by jkosternl**
Refactored mechanism how the EasyDoclet in the unittests was used. Now it uses less internals of the JVM, but uses the `Start` class of the Javadoc tools. The same tools are used by the Javadoc binary itself, so it is more future proof. 

Secondly, bumped a lot of plugins used and deps were available.

### CODE REVIEW

File: frank-doc-doclet/src/main/java/org/frankframework/frankdoc/doclet/Doclet.java [hunk 1]
```diff
@@ -54,6 +54,7 @@ class Doclet {
 				FrankElementFilters.getExcludeFilter(), FrankElementFilters.getExcludeFiltersForSuperclass());
 
 			model = FrankDocModel.populate(options.getDigesterRulesUrl(), options.getRootClass(), repository);
+			log.info("Found classes: {}", repository.size());
```
Comment (nielsm5) [inline frank-doc-doclet/src/main/java/org/frankframework/frankdoc/doclet/Doclet.java:57]:
```suggestion
			log.info("Found classes: {}", repository::size);
```

File: frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java [hunk 2]
```diff
@@ -1,87 +1,54 @@
...
 
-			// Invoke Javadoc and ask it for a DocletEnvironment containing the specified packages.
-			return javadocTool.getEnvironment(
-				options, // options
-				List.of(), // java names
-				List.of()); // java files
-		}
+	@Override
+	public SourceVersion getSupportedSourceVersion() {
+		return SourceVersion.RELEASE_11;
```
Comment (nielsm5) [inline frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java:43]: ?

File: frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java [hunk 3]
```diff
@@ -1,87 +1,54 @@
...
 	}
 
+	@Override
+	public boolean run(DocletEnvironment environment) {
+		log.debug("EasyDoclet.run start");
+		docTrees = environment.getDocTrees();
+		includedElements = environment.getIncludedElements();
+		log.debug("EasyDoclet.run: includedElements size = {}", includedElements.size());
```
Comment (nielsm5) [inline frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java:51]:
```suggestion
		log.debug("EasyDoclet.run: includedElements size = {}", includedElements::size);
```

File: frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java (hunk 2 above)
Comment (jkosternl) [reply frank-doc-doclet/src/test/java/org/frankframework/frankdoc/testdoclet/EasyDoclet.java:43]: Fixed, 17 it should be.

File: frank-doc-doclet/src/main/java/org/frankframework/frankdoc/doclet/Doclet.java (hunk 1 above)
Comment (jkosternl) [reply frank-doc-doclet/src/main/java/org/frankframework/frankdoc/doclet/Doclet.java:57]: If I do that, I get the warning: `'org.apache.logging.log4j.util.Supplier' is deprecated`, so I'll leave it for now.

**Pull request merged**